### PR TITLE
[WIP] : SSUI beginnings

### PR DESCRIPTION
**Pull request by sshveta**

### CODE REVIEW

**Comment by sshveta** [comment]
Please review

**Comment by dajoRH** [comment]
Lint report for commit b1cce68294611d34f3d45ef5874ed40716d14170:

`cfme/fixtures/pytest_selenium.py`:
- :grey_exclamation: Line 1047:17: **T001** *print statement found.*
- :grey_exclamation: Line 1048:17: **T001** *print statement found.*
- :grey_exclamation: Line 1051:17: **T001** *print statement found.*
- :grey_exclamation: Line 1052:17: **T001** *print statement found.*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: cfme/fixtures/pytest_selenium.py [hunk 1]
```diff
@@ -1043,7 +1043,14 @@ def force_navigate(page_name, _tries=0, *args, **kwargs):
...
         if not current_user:  # default to admin user
-            login.login_admin()
+            if(store.current_appliance.mode == "ssui"):
+                print "-----------"
+                print store.current_appliance.mode
+                login.ssui_login()
+            else:
+                print "-----------"
+                print store.current_appliance.mode
+                login.login_admin()
```
Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: I see the linter complaining about the print statements, it is much mure practical to use logger in these cases.

File: cfme/fixtures/pytest_selenium.py [hunk 2]
```diff
@@ -1043,7 +1043,14 @@ def force_navigate(page_name, _tries=0, *args, **kwargs):
 
     def _login_func():
         if not current_user:  # default to admin user
-            login.login_admin()
+            if(store.current_appliance.mode == "ssui"):
```
Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: Remove the parentheses, please.

File: cfme/tests/ssui/conftest.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+def revert_to_default():
+    store.current_appliance.mode = "default"
+    browser.quit()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def set_ssui_mode(request):
+    store.current_appliance.mode = "ssui"
+    browser.quit()
+    # request.addfinalizer(revert_to_default)
```
Comment (mfalesni) [inline cfme/tests/ssui/conftest.py]: I'd personally say to put this addfinalizer call right at the beginning of the fixture.

**Comment by dajoRH** [comment]
Lint report for commit 26ea828baafd99169db2432f27bbc39a0b043d8d:

`cfme/fixtures/pytest_selenium.py`:
- :grey_exclamation: Line 946:37: **S001** *found %s formatter*

`cfme/tests/ssui/test_login.py`:
- :bangbang: Line 5:1: **E303** *too many blank lines (3)*
- :bangbang: Line 7:101: **E501** *line too long (120 > 100 characters)*
- :bangbang: Line 8:101: **E501** *line too long (144 > 100 characters)*
- :heavy_exclamation_mark: Line 14:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6a3da37bbcdbc2e1e9ef2d225c72c247b2ceaca1:

`cfme/fixtures/pytest_selenium.py`:
- :grey_exclamation: Line 946:37: **S001** *found %s formatter*

`cfme/tests/ssui/test_login.py`:
- :bangbang: Line 4:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by sshveta** [comment]
Test requires a catalog item to be created from admin user and then logging in to SSUI to check if it shows. 
Also used time.sleep(5) at places as the UI needs sometime .
Please suggest alternative .

**Comment by mfalesni** [comment]
@sshveta We'll have to dig into the SSUI to find out anything that would need to be added to wait_for_ajax probably so we don't have to use time.sleep

**Comment by dajoRH** [comment]
Lint report for commit 04f72ad44af01bce8c721bdc03a2536201906166:

`cfme/fixtures/pytest_selenium.py`:
- :grey_exclamation: Line 946:37: **S001** *found %s formatter*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5bf6e006c759ec96f8c10d9c1626c00a1de3e69e:

`cfme/fixtures/pytest_selenium.py`:
- :grey_exclamation: Line 946:37: **S001** *found %s formatter*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9ca54e431e643b14521b78589eddd09b7e93fc6b:

`cfme/fixtures/pytest_selenium.py`:
- :grey_exclamation: Line 946:37: **S001** *found %s formatter*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: cfme/fixtures/pytest_selenium.py [hunk 4]
```diff
@@ -929,6 +929,41 @@ def __getitem__(self, item):
...
+        # 1: login_admin handles an alert or CannotContinueWithNavigation appears.
+        # 2: Everything should work. If not, NavigationError.
+        raise exceptions.NavigationError(page_name)
+
+    if "context" in kwargs:
+        if not isinstance(kwargs["context"], ContextWrapper) and isinstance(
+                kwargs["context"], dict):
+            kwargs["context"] = ContextWrapper(kwargs["context"])
+
+    _tries += 1
```
Comment (RonnyPfannschmidt) [inline cfme/fixtures/pytest_selenium.py]: i just noticed that tries doesn't seem to be used internally
i wonder if it should be factored out (it seems tobe the same in normal force navigate

@psav @mfalesni

File: cfme/fixtures/service_fixtures.py [hunk 5]
```diff
@@ -32,12 +32,8 @@ def catalog():
     catalog = "cat_" + fauxfactory.gen_alphanumeric()
     cat = Catalog(name=catalog,
                   description="my catalog")
-    try:
-        cat.create()
-        yield cat
-    finally:
-        if cat.exists:
-            cat.delete()
+    cat.create()
```
Comment (RonnyPfannschmidt) [inline cfme/fixtures/service_fixtures.py]: is there a reason why the removal is no longer necessary?

Comment (sshveta) [inline cfme/fixtures/service_fixtures.py]: Yes , I need  the catalog in Testing the links in SSUI .If the catalog is deleted the catalog item is no more visible in SSUI

**Comment by sshveta** [comment]
Closing this one and opening a new one

**Pull request closed without merging**